validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorRegistrator.java

```diff
@@ -177,28 +183,46 @@ private SafeFuture<Void> registerValidators(final List<Validator> validators) {
         .refresh()
         .thenCompose(
             __ -> {
-              final Stream<SafeFuture<SignedValidatorRegistration>> validatorRegistrationsFutures =
-                  createValidatorRegistrations(validators);
-              return SafeFuture.collectAllSuccessful(validatorRegistrationsFutures)
-                  .thenCompose(validatorRegistrationBatchSender::sendInBatches);
+              final SafeFuture<List<SignedValidatorRegistration>> validatorRegistrations =
+                  filterActiveValidators(validators)
+                      .thenCompose(this::createValidatorRegistrations);
+              return validatorRegistrations.thenCompose(
+                  validatorRegistrationBatchSender::sendInBatches);
             });
   }
 
-  private Stream<SafeFuture<SignedValidatorRegistration>> createValidatorRegistrations(
+  private SafeFuture<List<Validator>> filterActiveValidators(final List<Validator> validators) {
+    final Map<BLSPublicKey, Validator> validatorMap =
+        validators.stream().collect(Collectors.toMap(Validator::getPublicKey, Function.identity()));
+    return validatorApiChannel
+        .getValidatorStatuses(validators.stream().map(Validator::getPublicKey).toList())
```

Comment:
hopefully on a remote validator client this results in 1 api call?
It'd be worth setting up a BN/VC combo and just making sure we only do the 1, or we might end up doing 10k rest api calls, which would be not fun...

Reply:
it was one and I'm currently remaking it to do in batches, because it will fail with 5k validators

Reply:
might be worth having some timings around it if we don't... This might be why theres a lot of timeouts on goerli...

Reply:
given it doesnt do that now, happy to leave that out of this review :) might be a few bits in that refactor...

Reply:
I'd check how it works in this style, we will already get a gap between registration calls due to statuses query